Automattic/wp-calypso
# Backups: Display upsell for Atomic sites missing the needed plan

Fixes https://github.com/Automattic/wp-calypso/issues/51588
Requires D82943-code

#### Proposed Changes

Removes a few assumptions from the code that were considering that Atomic sites always had access to Jetpack Backup and ensures that upsells are displayed if they lack the required plan.

Before | After
--- | ---
<img width="780" alt="Screen Shot 2022-06-21 at 16 34 45" src="https://user-images.githubusercontent.com/1233880/174828488-70f1b47e-d15a-4f76-83e9-3c96c95ce0f6.png"> | <img width="759" alt="Screen Shot 2022-06-21 at 16 34 56" src="https://user-images.githubusercontent.com/1233880/174828522-44e79c5b-b301-45fd-a6c8-7a643433afb1.png">

Note that changes are easier to review when applying the unified view and hiding the whitespaces:

<img width="251" alt="Screen Shot 2022-06-22 at 14 53 30" src="https://user-images.githubusercontent.com/1233880/175033976-69c46488-d0c4-4df2-b2b9-ed09ab89cbd1.png">


#### Testing Instructions

- Apply D82943-code and sandbox the API.
- Go to SA and remove the plan from an Atomic site.
- Use the Calypso live link below and switch to that Atomic site.
- Go to Jetpack > Backup.
- Make sure you are asked to upgrade to the Pro plan.
- Make sure there are no regressions with other combinations:
  - WP.com site (Simple) + Free plan: Should display upsell
  - WP.com site (Simple) + Pro plan: Should display Atomic transfer banner
  - WP.com site (Atomic) + Pro plan: Should display latest backups
  - Jetpack site + Free plan: Should display upsell
  - Jetpack site + Backup plan: Should display latest backups
